arrow-ipc/src/reader.rs

```diff
@@ -399,6 +431,13 @@ pub fn read_record_batch(
     let field_nodes = batch.nodes().ok_or_else(|| {
         ArrowError::IpcError("Unable to get field nodes from IPC RecordBatch".to_string())
     })?;
+
+    let mut variadic_counts: VecDeque<i64> = if let Some(v) = batch.variadicBufferCounts() {
+        v.iter().collect()
+    } else {
+        VecDeque::default()
+    };
```

Comment:
(very minor) If you are into functional programming you can make this code more concise like this I think:

```suggestion
    let mut variadic_counts: VecDeque<i64> = batch.variadicBufferCounts()
        .map(|v| v.iter().collect())
        .unwrap_or_default();```

Reply:
```suggestion
    let mut variadic_counts: VecDeque<i64> = batch.variadicBufferCounts().iter().flatten().collect()
```

Or even